File: 0max-consecutive-ones/0max-consecutive-ones.py
class Solution(object):
    def findMaxConsecutiveOnes(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        N = len(nums)
        A = nums
        cnt = 0
        ans = 0
        for i in range(N):
            if (i == 0 or A[i] != A[i - 1]) and A[i] == 1:
                cnt = 1
            elif A[i] == 1:
                cnt += 1
            else:
                cnt = 0
            ans = max(ans, cnt)
        return ans

File: 0max-consecutive-ones/0max-consecutive-ones.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int max_n = 0;
        int n = 0;
        for(int i = 0; i < nums.length; i++) {
            if(nums[i] == 1) {
                n++;
            } else {
                max_n = Math.max(n, max_n);
                n = 0;
            }
        }
        max_n = Math.max(n, max_n);
        
        return max_n;
    }
}